src/extensions/replay/triggerMatching.ts

```diff
@@ -0,0 +1,314 @@
+import {
+    SESSION_RECORDING_EVENT_TRIGGER_ACTIVATED_SESSION,
+    SESSION_RECORDING_URL_TRIGGER_ACTIVATED_SESSION,
+} from '../../constants'
+import { PostHog } from '../../posthog-core'
+import { FlagVariant, RemoteConfig, SessionRecordingUrlTrigger } from '../../types'
+import { isBoolean, isObject, isString } from '../../utils/type-utils'
+import { isNullish } from '../../utils/type-utils'
+import { window } from '../../utils/globals'
+
+export type TriggerType = 'url' | 'event'
```

Comment:
I notice that in most of the cases such types are used instead of enums. I'm slightly triggered as raw strings like `trigger_activated` are used in checks and so, but it looks like it's safe enough because of TS validation. Plus, I assume these types are erased when building into the actual client script, so the size is lower?

No action is required, just thinking out loud :)

Reply:
yep... they're not really raw strings when typed like this so TS protects you

https://www.totaltypescript.com/why-i-dont-like-typescript-enums says it better than i could why i prefer this approach